File: Challenge1/Solutions/520. Detect Capital.js
/**
 * @param {string} word
 * @return {boolean}
 */
var detectCapitalUse = function(word) {
    let t,cap=0,small=0,firstcap=false,n=word.length;
    for(let i=0;i<n;i++){
        t=word[i];
        if(t.charCodeAt(0)>=65 && t.charCodeAt(0)<=90){
            if(i==0)
                firstcap=true;
            cap++;
        }
        else{
            small++;
        }
        
    }
    if(firstcap){
        if(n==(small+1))
            return true;
        if(n==cap)
            return true;
    }
    else if(n==small){
        return true;
    }
    return false;
    
};

File: Challenge1/Java Solutions/520 Detect Capital.java
class Solution {
    public boolean detectCapitalUse(String word) {
        
        Boolean firstCap=false;
        Boolean allCap=false;
        
        if(word.charAt(0)<='Z' && word.charAt(0)>='A')
            firstCap=true;
        
        if(word.length()==1)
            return true;
        
        if(word.charAt(1)<='Z' && word.charAt(1)>='A')
            allCap=true;
                    
        for(int i=1;i<word.length();i++){
            if(firstCap){
                if(allCap){  

                    if(word.charAt(i)>'Z' || word.charAt(i)<'A')
                        return false;
                }
                else if(word.charAt(i)<='Z' && word.charAt(i)>='A')
                    return false;     
            }
            else{
                if(word.charAt(i)<='Z' && word.charAt(i)>='A')
                    return false;
                }
        }
        return true;
    }
}
